app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
```

Comment:
1.Should return an Enumerator if no block is given. For example:
 `p [1, 2, 3].my_each` should return `#<Enumerator: [1, 2, 3]:my_each>`

2. When a block is given when self is an array should return the array itself after calling the given block once for each element in self. For example:
`[1,2,3,4,5].my_each { |num| num < 6}` should return `[1, 2, 3, 4, 5]`

3. When a block is given when self is a range should return the range itself after calling the given block once for each element in self. For example:
`(1..5).my_each { |num| num < 6}` should return `1..5`

4. When a block is given when self is a hash should behave like #each calls the given block once for each element in self. For example:
```ruby
test = {"a" => 1, "b" => 2, "c" => 3}
test.my_each{|key, value| puts "#{key}, #{value}" } 
```
should return 
```ruby
  a, 1 
  b, 2
  c, 3
```